File: rides-service/src/main/java/com/modsen/ride/client/DriverServiceClientFallback.java
package com.modsen.ride.client;

import com.modsen.ride.util.ExceptionMessages;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Component;

import java.util.UUID;

@Component
public class DriverServiceClientFallback implements DriverServiceClient {
    @Override
    public ResponseEntity<Boolean> doesDriverExists(UUID driverId) {
        throw new RuntimeException(ExceptionMessages.UNABLE_TO_REACH_DRIVER_SERVICE.format());
    }
}

File: rating-service/src/main/kotlin/com/modsen/rating/client/DriverServiceClientFallback.kt
package com.modsen.rating.client

import com.modsen.rating.util.ExceptionMessages
import java.util.UUID
import org.springframework.http.ResponseEntity
import org.springframework.stereotype.Component

@Component
class DriverServiceClientFallback: DriverServiceClient {
    override fun doesDriverExist(driverId: UUID): ResponseEntity<Boolean> {
        throw RuntimeException(ExceptionMessages.UNABLE_TO_REACH_DRIVER_SERVICE)
    }
}